cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -237,13 +247,58 @@ private void processTetheringControlResponse(String message, PeerInfo peerInfo)
   }
 
   /**
-   * Persist received TetheringControlMessages to {@link Constants.Tethering#TETHERING_TOPIC}
+   * Modify received files to reference the namespace in this instance and to include the peer name for RunRecord.
+   * Use these files to start the program through ProgramStateWriter.
+   * Save the rest of the files into a temp dir (to be loaded when the Program is created in DistributedProgramRunner).
    */
-  private void publishTetheringControlMessage(TetheringControlMessage message) throws IOException {
-    try {
-      messagePublisher.publish(topic.getNamespace(), topic.getTopic(), StandardCharsets.UTF_8, GSON.toJson(message));
-    } catch (TopicNotFoundException | IOException e) {
-      throw new IOException(String.format("Failed to publish message to topic %s", topic.getTopic()), e);
+  private void publishRunPipeline(String runPayload, String peerName) throws IOException {
```

Comment:
RunPipeline -> StartProgram. 

Running is a separate program state and pipelines are a type of program